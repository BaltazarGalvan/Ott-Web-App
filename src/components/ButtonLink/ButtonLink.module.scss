@use '../../styles/variables';
@use '../../styles/theme';
@use '../../styles/mixins/responsive';
@use '../../styles/mixins/utils';

//
// Navigation Button
// --------------------------------

.link {
  position: relative;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  min-height: 38px;
  padding: 0 10px;
  color: inherit;
  font-family: theme.$body-alt-font-family;
  font-weight: 700;
  text-align: center;
  text-decoration: none;
  background: transparent;
  border-radius: 4px;
  cursor: pointer;
  opacity: 0.7;
  transition: background 0.1s ease, transform 0.1s ease;

  @media (hover: hover) and (pointer: fine) {
    &:hover,
    &:active {
      z-index: 1;
      background: rgba(variables.$black, 0.7);
      transform: scale(1.1);
      opacity: 1;
    }
  }
}

.active {
  color: variables.$white;
  opacity: 1;
}
