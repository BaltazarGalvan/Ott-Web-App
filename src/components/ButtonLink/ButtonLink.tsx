import React from 'react';
import { NavLink } from 'react-router-dom';

import styles from './ButtonLink.module.scss';

type ButtonLinkProps = {
  label: string;
  to: string;
};

const ButtonLink: React.FC<ButtonLinkProps> = ({
  label,
  to,
}: ButtonLinkProps) => {
  return (
    <NavLink
      className={styles.link}
      activeClassName={styles.active}
      to={to}
      exact
    >
      <span className={styles.buttonLabel}>{label}</span>
    </NavLink>
  );
};

export default ButtonLink;
