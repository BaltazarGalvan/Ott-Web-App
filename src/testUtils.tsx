import { BrowserRouter as Router } from 'react-router-dom';
import React, { ReactNode, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

interface WrapperProps {
  children?: ReactNode;
}

export const wrapper = ({ children }: WrapperProps) => <Router>{children}</Router>;

const customRender = (ui: ReactElement, options?: RenderOptions) => render(ui, { wrapper, ...options });

export const mockWindowLocation = (path: string) => {
  Object.defineProperty(window, 'location', {
    value: {
      pathname: path,
      assign: jest.fn(),
    },
  });
};

export const mockMatchMedia = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    })),
  });
};

export { customRender as render };
