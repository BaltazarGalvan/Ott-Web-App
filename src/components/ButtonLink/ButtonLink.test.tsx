import React from 'react';

import { render } from '../../testUtils';

import ButtonLink from './ButtonLink';

describe('<ButtonLink />', () => {
  test('renders link with route label', () => {
    const { getByText } = render(<ButtonLink label="home" to="/" />);
    expect(getByText(/home/i)).toBeTruthy();
  });

  it('renders and matches snapshot', () => {
    const { container } = render(<ButtonLink label="home" to="/" />);

    expect(container).toMatchSnapshot();
  });
});
