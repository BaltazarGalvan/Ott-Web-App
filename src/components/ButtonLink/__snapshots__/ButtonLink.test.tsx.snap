// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLink /> renders and matches snapshot 1`] = `
<div>
  <a
    aria-current="page"
    class="function link() { [native code] } active"
    href="/"
  >
    <span>
      home
    </span>
  </a>
</div>
`;
